Projects/Embark/Sources/EmbarkState+Tracking.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import hGraphQL
+import hCore
+
+public struct EmbarkTrackingEvent {
+    public var title: String
+    public var properties: [String:Any]
+    
+    func send() {
+        Self.trackingHandler(self)
+    }
+}
+
+public extension EmbarkTrackingEvent {
+    static var trackingHandler: (_ Event: EmbarkTrackingEvent) -> Void = { _ in }
+}
+
+internal extension GraphQL.EmbarkExternalRedirectLocation {
+    func trackingEvent(store: EmbarkStore) -> EmbarkTrackingEvent {
+        var properties = [String:String]()
+        if let store = store.revisions.last {
```

Comment:
maybe we should have a `func getAllValues() -> [String: String]` and that one just goes into revisions and picks the last one. what you think?